lib/adapter.js

```diff
@@ -1070,10 +1070,26 @@ function Adapter(options) {
         if (state !== undefined) {
             delete task.state;
         }
-        adapterObjects.extendObject(task._id, task, () =>
-            state ?
-                adapterStates.setState(task._id, state, () => setImmediate(extendObjects, tasks, callback)) :
-                setImmediate(extendObjects, tasks, callback));
+        adapterObjects.getObject(task._id, (err, obj) => {
```

Comment:
I think you can simplify here. I would do the "getObject" to find out if it is existing already. Then always call "extendObject" (because behaves like "setObject" for non existing objects) and then check in one place if obj was already there && state !== undefined.

So you should not need that if/else with very comparable code